### PR TITLE
[ENHANCEMENT] `argilla`: Add limit argument when getting records

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds a new `limit` argument when getting records from the Argilla server. 

**Type of change**

- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/src/argilla/records/_dataset_records.py [hunk 1]
```diff
@@ -62,19 +63,37 @@ def __init__(
         self.__with_responses = with_responses
         self.__with_vectors = with_vectors
         self.__records_batch = []
+        self.__limit = 1 if limit == 0 else limit
```
Comment (davidberenstein1957) [inline argilla/src/argilla/records/_dataset_records.py]:
```suggestion
        self.__limit = 1 if limit == 0 else limit
```
```suggestion
        self.__limit = 1 if limit <= 0 else limit
```

Comment (davidberenstein1957) [reply argilla/src/argilla/records/_dataset_records.py]: should we provide a warning if it is 0 or lower than 0?

File: argilla/tests/integration/test_list_records.py [hunk 2]
```diff
@@ -66,6 +66,77 @@ def test_list_records_with_start_offset(client: Argilla, dataset: Dataset):
     ]
 
 
+def test_list_records_with_limit(client: Argilla, dataset: Dataset):
```
Comment (davidberenstein1957) [inline argilla/tests/integration/test_list_records.py:69]: perhaps we should add test cases for limit extremes, being 0 or negative.

**Review by burtenshaw — APPROVED** [review]
